packages/jest-runner/fuzz.ts

```diff
@@ -64,15 +68,14 @@ export const fuzz: FuzzTest = (name, fn) => {
 	// points to the element containing the fuzz function.
 	const testStatePath = currentTestStatePath(testName);
 
-	// The timeout setting is extracted from the test state and defaults to 5s.
-	// Setting timeouts on this level is necessary, as they apply to the
-	// individual inputs and not the whole run.
-	const timeout = currentTimeout();
-
 	const corpus = new Corpus(testFile, testStatePath);
 
 	const fuzzingConfig = loadConfig();
-	fuzzingConfig.timeout = timeout;
+
+	// Timeout priority is: test timeout > config timeout > default timeout.
+	if (!timeout)
```

Comment:
Could be moved into `runInRegressionMode`, as it's not loaded from the global config anymore and only relevant for that case.

Reply:
It cannot, because the config needs to be updated based on the given timeout and forwarded to the `runInFuzzingMode`, if you accept my reasoning regarding the "timeout" flag above!

Reply:
Looks good.